src/server/AppManager.ts

```diff
@@ -733,6 +735,31 @@ export class AppManager {
         return this.enableApp(storageItem, app, true, isManual, silenceStatus);
     }
 
+    private async installApp(storageItem: IAppStorageItem, app: ProxiedApp, user: IUser): Promise<boolean> {
+        let result: boolean;
+        const read = this.getAccessorManager().getReader(storageItem.id);
+        const http = this.getAccessorManager().getHttp(storageItem.id);
+        const persistence = this.getAccessorManager().getPersistence(storageItem.id);
+
+        try {
+            await app.call(AppMethod.ONINSTALL, read, http, persistence, user);
+
+            result = true;
+        } catch (e) {
+            const status = AppStatus.ERROR_DISABLED;
+
+            if (e.name === 'NotEnoughMethodArgumentsError') {
+                console.warn('Please report the following error:');
```

Comment:
Don't simply use `console.warn`, use `app.getLogger().warn` instead. Please append the original error to the log as well.

Reply:
Ok I've fixed it. I used `console.warn` because other methods in this file use it, my bad!